tools/common/__init__.py

```diff
@@ -192,8 +192,11 @@ async def update_outbound_ws(
 async def wait_for_keyboard_interrupt() -> None:
     """Wait for keyboard interrupt (Ctrl-C)."""
     sig_event = asyncio.Event()
-    signal.signal(signal.SIGINT, lambda _exit_code, _frame: sig_event.set())
+    loop = asyncio.get_running_loop()
+    loop.add_signal_handler(signal.SIGINT, lambda: sig_event.set())
     await sig_event.wait()
+    # In a real program, you may want to be more graceful here
+    sys.exit(1)
```

Comment:
This was not here by intention to allow a clean shutdown which we also use to verify when we use the example:
https://github.com/home-assistant-libs/aioshelly/blob/bf970c5decf5e1125978274d0c8c3d0295921901/tools/example.py#L47-L48
```suggestion
```